cpp/src/cylon/table.hpp

```diff
@@ -204,6 +198,17 @@ class Table {
    */
   std::vector<std::shared_ptr<cylon::Column>> GetColumns() const;
 
+#ifdef BUILD_CYLON_PARQUET
+  /**
+  * Write the table as a parquet file
+  * @param path file path
+  * @return the status of the operation
+  */
```

Comment:
Shall we put all the parquet related methods under one check?

Reply:
In a single #Ifdef ... #endif segment in the file? The issue is that WriteParquet is a part of Table's public functions while the reads are not. Or do you mean to put all the parquet related code into separate files and only check in table_api? (like table_api calling from parquet.hpp rather than table.hpp)

Reply:
Got it. I think I missed that part